src/basic/socket-util.c

```diff
@@ -1226,3 +1241,142 @@ int socket_pass_pktinfo(int fd, bool b) {
                 return -EAFNOSUPPORT;
         }
 }
+
+int socket_pass_err(int fd, int af, bool b) {
```

Comment:
Hmm, why is this called "pass_err" not "set_recverr"? Can we call the functions in a way that simply matches the socket option instead of having "independent" naming?